libs/utils/src/mutex.ts

```diff
@@ -29,17 +31,27 @@ export class Mutex<T> {
     }
 
     this.locked = true;
+    const { value } = this;
     let unlockCalled = false;
 
     return {
-      value: this.value,
+      get value(): T {
+        if (unlockCalled) {
+          throw new Error('value accessed after unlock');
+        }
+
+        return value as T;
+      },
+
       unlock: () => {
         if (unlockCalled) {
           throw new Error('unlock called more than once');
         }
         unlockCalled = true;
         this.locked = false;
-        for (const resolve of this.asyncQueue) {
+
+        while (!this.locked && this.asyncQueue.length > 0) {
+          const resolve = assertDefined(this.asyncQueue.shift());
           resolve();
         }
```

Comment:
So one more thing that I'm wondering is, in the `asyncLock` function below:

https://github.com/votingworks/vxsuite/blob/52933c0c9e56e577515e48eece72da8a7aed2809/libs/utils/src/mutex.ts#L53-L66

Do we need to move the creation of the `deferred` promise into the `for` loop? I'm reasoning through the following case:

- Entity 0 has the lock, and entities 1 and 2 are waiting to acquire the lock.
- Entity 0 releases the lock, and the blocking promises are resolved.
- Entity 1 acquires the lock.
- Entity 2 fails to acquire the lock and calls `this.asyncQueue.push(resolve)`.
- But the `await promise` doesn't pause execution because the promise was already resolved.
- Infinite loop until entity 1 releases the lock.

I'm still confirming the behavior of `deferred` but thought I'd run the question by you while I verify, myself